### PR TITLE
Replace Y.Array with utils

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add two charms
- relate them
- you should not have any errors in the console

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7129/

File: jujugui/static/gui/src/app/views/utils.js [hunk 1]
```diff
@@ -1925,6 +1925,50 @@ YUI.add('juju-view-utils', function(Y) {
...
+  /**
+    Remove duplicate entries from an array.
+
+    @method arrayDedupe
+    @returns {Array} An array with no duplicates.
+  */
+  utils.arrayDedupe = function(array) {
+    // Sets can only contain unique values, so use that to do the dedupe and
+    // then turn it back into an array.
+    return Array.from(new Set(array));
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: I'm not sure if this is any more performant....

```JavaScript
[...new Set(array)]
```

File: jujugui/static/gui/src/app/views/utils.js [hunk 2]
```diff
@@ -1925,6 +1925,50 @@ YUI.add('juju-view-utils', function(Y) {
...
+        // the current value.
+        Array.isArray(current) ? utils.arrayFlatten(current) : current);
+    }, []);
+  };
+
+  /**
+    Map two arrays into an array of pairs for each position from the original
+    arrays e.g. [1, 2] and [3, 4] would become [[1, 3], [2, 4]]
+
+    @method arrayFlatten
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: arrayZip

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7160/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**